src/core/surface/call.c

```diff
@@ -425,7 +425,13 @@ static grpc_cq_completion *allocate_completion(grpc_call *call) {
     if (call->allocated_completions & (1u << i)) {
       continue;
     }
-    call->allocated_completions |= (gpr_uint8)(1u << i);
+    /* NB: the following integer arithmetic operation needs to be in its
+     * expanded form due to the "integral promotion" performed (see section
+     * 3.2.1.1 of the C89 draft standard), which in this case upcasts the result
+     * of the bitwise OR to "unsigned". A cast to the smaller container type is
```

Comment:
We are talking about different things. I think you are talking about call->allocated_completions being promoted to int prior to the bitwise or. I'm talking about the whole bitwise or expression in parenthesis, whose result is an unsigned because of of its operands is unsigned, namely (1u << i).